api/lib/models/project/project-data-mapper-impl.ts

```diff
@@ -0,0 +1,180 @@
+    const query = squel.delete({ autoQuoteTableNames: true, autoQuoteFieldNames: true })
+      .from(this.tableName)
+      .where(`${this.pkColumnName} = ?`, projectID)
+      .toParam();
+    query.text = query.text.concat(';');
+    return from(
+      this.sql.query(query.text, query.values, { stream: false, cache: false }),
+    ).pipe(
+      map(() => ({ result: 'Success', data: undefined })),
+    ).toPromise();
+  }
+
+  public get(projectID: string, opts?: IUowOpts): Promise<IDbResult<Project>> {
+    let queryBuilder = squel.select({ autoQuoteFieldNames: true, autoQuoteTableNames: true })
+      .from(this.tableName);
+    if (opts && opts.fields) {
+      queryBuilder = queryBuilder.fields(opts.fields);
+    }
+    queryBuilder = queryBuilder
+      .where(`${this.pkColumnName} = ?`, projectID);
+    const query = queryBuilder.toParam();
+    query.text = query.text.concat(';');
+    return from(this.sql.query<Project>(
+      query.text,
+      query.values,
+      { stream: false, cache: true },
+    ))
+      .pipe(
+        map((event: Project[]) => ({ result: 'Success', data: event[0] })),
+      )
+      .toPromise();
+  }
+
+  public async getAll(opts?: IUowOpts): Promise<IDbResult<Stream<Project>>> {
+    throw new MethodNotImplementedError('This method is not implemented');
+  }
+
+  public async getCount(opts?: IUowOpts): Promise<IDbResult<number>> {
+    let queryBuilder = squel.select({ autoQuoteTableNames: true, autoQuoteFieldNames: false })
+      .from(this.tableName)
+      .field(`COUNT(${this.pkColumnName})`, 'rsvp_count');
+    if (opts && opts.byHackathon) {
+      queryBuilder = queryBuilder
+        .where(
+          'hackathon = ?',
+          await (opts.hackathon ?
+            Promise.resolve(opts.hackathon) :
+            this.activeHackathonDataMapper.activeHackathon
+              .pipe(map(hackathon => hackathon.uid))
+              .toPromise()),
+        );
+    }
+    const query = queryBuilder
+      .toString()
+      .concat(';');
+    return from(
+      this.sql.query<number>(query, [], { stream: true, cache: true }),
+    ).pipe(
+      map((result: number) => ({ result: 'Success', data: result })),
+    ).toPromise();
+  }
+
+  // need to be checked for conversion
+  public async insert(object: Project): Promise<IDbResult<Project>> {
+    const validation = object.validate();
+    if (!validation.result) {
+      this.logger.warn('Validation failed while adding object.');
+      this.logger.warn(object.dbRepresentation);
+      return Promise.reject({ result: 'error', data: new HttpError(validation.error, 400) });
+    }
+    let query = 'CALL ';
+    query = query.concat('assignTeam');
+    query = query.concat('(?,?,?,@projectID_out); SELECT @projectID_out as projectID;')
+    const list = [object.project_name, object.team.join(','), object.categories.join(',')];
+    return from(
+      this.sql.query<void>(query, list, { stream: false, cache: false }),
```

Comment:
Since this runs a `SELECT projectID_out as projectID;`, the query will end up returning `[projectID]`